### ISSUE TITLE
Troubles using cheerio in custom Webpack loader

### ISSUE THREAD

**Issue by mehdi-cit** [cheeriojs/cheerio#1021]
Hello and thank you for the wonderful Cheerio.
I'm trying to use it in a custom webpack loader for some react project ( this loader loads js and jsx files and uses cheerio to manipulate them prior to running the babel loader on them).
Cheerio is great and very useful except that it alters the JSX/HTML in ways that cause problems.
For instance, I load in it some JSX 'string' in it such: `<cp1 key={X}>One</cp1>`

The above is valid JSX, however cheerio alters it into  `<cp1 key="{X}">One</cp1>`
In short it tries to be smart and quote the key attribute as if it where HTML. But when it does that it renders my JSX obsolete as the key attribute will hold the string value "{X}" instead of evaluating  `{X}`

Is there a way to turn off any such alterations ? (In my loader i need to explicitly change the code and not having to deal with cheerio's implicit alterations).

**Comment by mehdi-cit** [comment]
Hello,
For anyone stumbling upon a similar problem, you may want to look as function ``function formatAttrs(attributes, opts) {`
In npm module 'dom-serializer' which is one of cheerio's dependencies. More specifically the line below:
```` output += key + '="' + (opts.decodeEntities ? entities.encodeXML(value) : value) + '"';

**Comment by stevenvachon** [comment]
JSX is not HTML.

**Comment by fb55** [comment]
Sadly we only target HTML, JSX will need at least custom serialisation logic :/
